### PR TITLE
Provisioning: Provision dashboards into subfolders

### PR DESCRIPTION
**Pull request by undef1nd**
This change enables the provisioning of dashboards into subfolders when the nested folders feature is on.
Please note the adjustment for folder titles with slashes: ensure that they are properly escaped when specified in the provisioning configuration.

--------------------
# Release notice breaking change

Users who provision folders with slashes in the title from now on have to escape them:

for example, if dashboard provisioning config contains:
`folder: folder_with_/_in_title`
it must be changed to:
`folder: folder_with_\/_in_title`

### CODE REVIEW

File: pkg/services/folder/folderimpl/folder.go [hunk 1]
```diff
@@ -555,7 +557,9 @@ func (s *Service) Create(ctx context.Context, cmd *folder.CreateFolderCommand) (
 	dashFolder := dashboards.NewDashboardFolder(cmd.Title)
 	dashFolder.OrgID = cmd.OrgID
 
-	if s.features.IsEnabled(ctx, featuremgmt.FlagNestedFolders) && cmd.ParentUID != "" {
+	// TODO: This feature flag check causes issue for provisioning: folder is created without FolderUID in dashboards table
+	// Then provisioning getOrCreateFolderByTitle fails to find the folder and tries to create it again even thought it already exists
```
Comment (undef1nd) [inline pkg/services/folder/folderimpl/folder.go:561]: @papagian Sofia, I need your help with deciding if it's the right thing to do. Suggest to discuss it offline.

File: pkg/services/folder/folderimpl/folder.go [hunk 2]
```diff
@@ -555,7 +557,7 @@ func (s *Service) Create(ctx context.Context, cmd *folder.CreateFolderCommand) (
 	dashFolder := dashboards.NewDashboardFolder(cmd.Title)
 	dashFolder.OrgID = cmd.OrgID
 
-	if s.features.IsEnabled(ctx, featuremgmt.FlagNestedFolders) && cmd.ParentUID != "" {
```
Comment (undef1nd) [inline pkg/services/folder/folderimpl/folder.go:558]: When the feature flag is disabled and provisioning dashboards in nested folders, a folder is stored without parent folder_uid in `dashboards` table.

So when provisioning service checks by title & folder_uid if such a folder exists (in `getOrCreateFolderByTitle` method), it fails to find a match and attempts to create it again even thought it already exists.

Comment (papagian) [reply pkg/services/folder/folderimpl/folder.go:558]: I'm leaning towards it's better to not create nested folders if the flag is not set therefore I would suggest to better hide my changes behind the feature toggle.

Comment (undef1nd) [reply pkg/services/folder/folderimpl/folder.go:558]: draft for this is in branch https://github.com/grafana/grafana/tree/provisioning-with-featToggle

**Comment by QuentinBisson** [comment]
Hey @undef1nd and @papagian do you know if you will work on this feature? This is really needed by some users, me included, who cannot use nested folders until this is solved

**Comment by papagian** [comment]
> Hey @undef1nd and @papagian do you know if you will work on this feature? This is really needed by some users, me included, who cannot use nested folders until this is solved. Is there any way I could help?

sorry, It has just fallen behind other priorities: I will have a fresh look

**Comment by QuentinBisson** [comment]
:(

**Comment by papagian** [comment]
> We will have to migrate from the side-car approach then because it's unsupported :(

I'm sorry, I hadn't forget about that but I didn't get time to fix the tests

**Pull request closed without merging**